index.html

```diff
@@ -54,59 +49,98 @@
+<h2><dfn>WorkerNavigator</dfn> partial interface</h2>
+<pre class="idl">
+[Exposed=DedicatedWorker, SecureContext]
+partial interface WorkerNavigator {
+  [SameObject] readonly attribute Serial serial;
 };
 </pre>
+<section>
+<h2><dfn>serial</dfn> attribute</h2>
 
-### Constructing `SerialPort` objects
-When the constructor of the `SerialPort` interface is invoked, the user agent MUST run the following steps:
+The <a>serial</a> attribute SHALL be an instance of the <a>Serial</a>
+interface.
+</section>
+</section>
 
-1. ...To be written...
-1. Profit!
-1. Return <var>serial port</var>.
+<section data-dfn-for="Serial" data-link-for="Serial">
+<h2><dfn>Serial</dfn> interface</h2>
+<pre class="idl">
+[Exposed=(DedicatedWorker,Window), SecureContext]
+interface Serial : EventTarget {
+  attribute EventHandler onconnect;
+  attribute EventHandler ondisconnect;
+  Promise&lt;sequence&lt;SerialPort>> getPorts();
+  [Exposed=Window] Promise&lt;SerialPort> requestPort(SerialPortRequestOptions options);
+};
+</pre>
 
-### `requestPorts()` method
+<section>
+<h2><dfn>requestPort()</dfn> method</h2>
 
-The `requestPorts()` method requests access to the serial ports of the system
-(see <a>security considerations</a>).
+The <a>requestPort()</a> method requests access to a serial port of the user's
+choice (see <a>security considerations</a>).
 
-When the `requestPorts()` method is invoked, the user agent MUST perform the
+When the <a>requestPort()</a> method is invoked, the user agent MUST perform the
 following steps:
 
-1. Let <var>promise</var> be a new <a>`Promise`</a> object and <var>resolver</var>
-   its associated resolver.
-1. Return `promise` and run the remaining steps asynchronously.
-1. Let <var>ports</var> be an empty `Array` object [[!ECMASCRIPT]].
-1. Request access to serial ports (see <a>security considerations</a>):
-  1. If the access request is rejected:
-    1. Let <var>error</var> be a new <a>`DOMException`</a> whose name is
-       `[AbortError](http://dom.spec.whatwg.org/#aborterror)`.
-    1. Run resolver's internal reject algorithm with <var>error</var> as value
-       and terminate this algorithm.
-1. Otherwise, for each <var>port</var> for which access was granted:
-  1. Let <var>info</var> be the result of running the <a>steps for getting
-       metadata of a serial port</a>.
-  1. Append <var>info</var> to <var>ports</var>.
-1. Run <var>resolver</var>'s internal fulfill algorithm with <var>ports</var>
-   as value.
-
-### `getInfo()` method
-
-The `getInfo()` method provides a means to get metadata corresponding to a
-`SerialPort` object.
-
-When the `getInfo()` method is invoked, the user agent MUST perform the
-<a>steps for getting metadata of a serial port</a>.
+ 1. Let <var>promise</var> be a newly created <a>`Promise`</a>.
+ 1. Return <var>promise</var> and run the remaining steps asynchronously.
+ 1. Request access to a serial port (see <a>security considerations</a>):
```

Comment:
Just a. suggestion, but maybe in a new PR, "request access to a serial port" could be formally defined in the Security Considerations section.

Reply:
I will tackle that in a new PR.